### PR TITLE
Attempt 2: Queen Psy Points Construction Permissions, Now With Anti-Grief!

### PR DESCRIPTION
**Pull request by KornFlaks**
## About The Pull Request

Complete refactor of old Queen Permissions PR. Reopening as a new PR for new review.

This PR proposes a queen only verb that mirrors the now defunct slashing permissions except with psy points usage by the rest of the hive. It gives the queen player options to allow the construction of silos and/or turrets by others in the hive that have the ability to do so (shrikes and hivelords).

Without a queen, it is unrestricted (as in anyone can build). Upon queen ascension, psy points usage becomes forbidden (only the queen can build silos and turrets). There is an option in the queen's Alien tab to switch modes. Upon queen death, it returns to unrestricted. Importantly, during periods without a silo, queens can not restrict building of one.

**Icon is currently a placeholder (using Psychic Whisper) and is intended to be hidden in the future for use in Hive Management Panel.**

## Why It's Good For The Game

A hivelord can singlehandedly ruin the game for an entire hive by wasting psy points on a silo just outside of FOB and there is nothing the queen or other players can do to stop them. AHelping for a refund or movement of the silo is a bandaid fix that depends on the generosity of the admins responding.

This implements an in-game solution by the leader of the hive. This of course now shifts the responsibility of managing the economy of the hive's psy points to the queen, regardless of actual skill. However, those evolving to queen should already understand that there are immense responsibilities in managing the hive upon taking on the role.

## Changelog
:cl:
add: Added the option for xeno Queens to allow / forbid the creation of psy points consuming constructions (silo and turrets) by others in the hive.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/xenoprocs.dm [hunk 1]
```diff
@@ -225,6 +225,17 @@
...
+
+	if(CHECK_BITFIELD(SSticker.mode.flags_round_type, MODE_PSY_POINTS))
+		if(hive.building_allowed == NONE)
+			stat("Expenditure of psy points status:", "FORBIDDEN")
+		else if(hive.building_allowed == XENO_BUILDING_SILO)
+			stat("Expenditure of psy points status:", "ONLY SILOS")
+		else if(hive.building_allowed == XENO_BUILDING_TURRET)
+			stat("Expenditure of psy points status:", "ONLY TURRETS")
+		else
+			stat("Expenditure of psy points status:", "UNRESTRICTED")
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/xenoprocs.dm:237]: use a switch

File: code/modules/mob/living/carbon/xenomorph/castes/queen/queen.dm [hunk 2]
```diff
@@ -100,8 +99,17 @@
...
+	if(!CHECK_BITFIELD(SSticker.mode.flags_round_type, MODE_PSY_POINTS))
+		return
 
+	// Checks if there are any building restrictions so there is no redundant declaration to the hive.
+	if(hive.building_allowed == XENO_BUILDING_UNRESTRICTED)
+		return
+	
+	// Resets building permissions to unrestricted upon queen death.
+	hive.building_allowed = XENO_BUILDING_UNRESTRICTED
+	xeno_message("With the death of the Queen, we are free from any restrictions on building silos or turrets.")
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/castes/queen/queen.dm:112]: like tivi said, maybe just show this if this was restricted?

**Review by BraveMole — APPROVED** [review]
Maybe put down a button until you make the hive management screen, that doesn't seem that awfull

File: code/modules/mob/living/carbon/xenomorph/castes/queen/queen.dm (hunk 2 above)
Comment (KornFlaks) [reply code/modules/mob/living/carbon/xenomorph/castes/queen/queen.dm:112]: It should be only shown when restricted. Did I accidently delete the check?

Comment (BraveMole) [reply code/modules/mob/living/carbon/xenomorph/castes/queen/queen.dm:112]: No, i'm just blind

File: code/modules/mob/living/carbon/xenomorph/xenoprocs.dm (hunk 1 above)
Comment (KornFlaks) [reply code/modules/mob/living/carbon/xenomorph/xenoprocs.dm:237]: No clue how I missed that. Done.

**Comment by KornFlaks** [comment]
Hold on a minute. Merge conflict broke something. Fuck.

**Pull request closed without merging**